src/main/java/com/pretz/dyntreeserver/service/exceptions/AuthExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package com.pretz.dyntreeserver.service.exceptions;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class AuthExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value = UserNotFoundException.class)
+    protected ResponseEntity<Object> handleUserNotFoundException(UserNotFoundException uEx) {
+        return new ResponseEntity<Object>("User not found.", new HttpHeaders(), HttpStatus.UNAUTHORIZED);
```

Comment:
I think putting generic type here is not needed? (`return new ResponseEntity<>(...);`)

Reply:
Changed